homeassistant/components/webostv/helpers.py

```diff
@@ -55,16 +56,18 @@ def async_get_client_by_device_entry(
     Raises ValueError if client is not found.
     """
     for config_entry_id in device.config_entries:
-        if entry := hass.config_entries.async_get_entry(config_entry_id):
-            client = entry.runtime_data
-            break
-
-    if not client:
-        raise ValueError(
-            f"Device {device.id} is not from an existing {DOMAIN} config entry"
-        )
-
-    return client
+        entry = hass.config_entries.async_get_entry(config_entry_id)
+        if entry and entry.domain == DOMAIN:
+            if entry.state is ConfigEntryState.LOADED:
+                return entry.runtime_data
+
+            raise ValueError(
+                f"Device {device.id} is not from a loaded {DOMAIN} config entry"
+            )
+
+    raise ValueError(
+        f"Device {device.id} is not from an existing {DOMAIN} config entry"
+    )
```

Comment:
I am also wondering if device triggers also work with HomeAssistantError, as we could add translations for these strings

Reply:
Oh but I don't necessarily mean we should add translation directly, I am more looking at ValueError vs HomeAssistantError

Reply:
Ok, understood, I don't have any preference, just note that this is an internal error not directly shown to the user, as it will be caught at:
https://github.com/home-assistant/core/blob/cccaa1e209ebda19ad08b51b706438fde507ca31/homeassistant/components/webostv/device_trigger.py#L46-L47 and raise `InvalidDeviceAutomationConfig`, if you think it is better to first raise `HomeAssistantError` I will change it.

Reply:
Ooh, I thought the validation code was part of core, not in the integration. In that case I think this is good